crates/polars-plan/src/plans/optimizer/simplify_expr.rs

```diff
@@ -440,6 +440,63 @@ impl OptimizationRule for SimplifyExprRule {
         let expr = expr_arena.get(expr_node).clone();
 
         let out = match &expr {
+            // drop_nulls().len() -> null_count()
+            // drop_nulls().count() -> null_count()
+            AExpr::Agg(IRAggExpr::Count(input, _)) => {
+                let input_expr = expr_arena.get(*input);
+                match input_expr {
+                    AExpr::Function {
+                        input,
+                        function: FunctionExpr::DropNulls,
+                        options: _,
+                    } => {
+                        let expr_ir = ExprIR::from_node(expr_node, expr_arena);
+                        let expr_output_name = expr_ir.output_name();
+                        let inner_minus_exp = AExpr::BinaryExpr {
+                            op: Operator::Minus,
+                            right: expr_arena.add(make_null_count_expr!(input)),
+                            left: expr_arena.add(AExpr::Len),
+                        };
+                        let inner_minus_node = expr_arena.add(inner_minus_exp);
+                        Some(AExpr::Alias(
+                            inner_minus_node,
+                            ColumnName::from(expr_output_name),
+                        ))
+                    },
+                    _ => None,
+                }
+            },
+            // is_null().sum() -> null_count()
+            // is_not_null().sum() -> len() - null_count()
+            AExpr::Agg(IRAggExpr::Sum(input)) => {
+                let input_expr = expr_arena.get(*input);
+                match input_expr {
+                    AExpr::Function {
+                        input,
+                        function: FunctionExpr::Boolean(BooleanFunction::IsNull),
+                        options: _,
+                    } => Some(make_null_count_expr!(input)),
+                    AExpr::Function {
+                        input,
+                        function: FunctionExpr::Boolean(BooleanFunction::IsNotNull),
+                        options: _,
+                    } => {
+                        let expr_ir = ExprIR::from_node(expr_node, expr_arena);
+                        let expr_output_name = expr_ir.output_name();
+                        let inner_minus_exp = AExpr::BinaryExpr {
+                            op: Operator::Minus,
+                            right: expr_arena.add(make_null_count_expr!(input)),
+                            left: expr_arena.add(AExpr::Len),
+                        };
+                        let inner_minus_node = expr_arena.add(inner_minus_exp);
+                        Some(AExpr::Alias(
```

Comment:
The alias shouldn't be needed here - what you should do is to use `AExpr::Count(input, true)` instead of `AExpr::Len` - that way the column should automatically be propagated to the output.